includes/Core/Tags/Google_Tag_Gateway/Google_Tag_Gateway_Settings.php

```diff
@@ -101,10 +154,16 @@ function ( $value ) {
 			'isEnabled'             => true,
 			'isGTGHealthy'          => true,
 			'isScriptAccessEnabled' => true,
+			'isGTGDefault'          => true,
 		);
 
 		$updated = array_intersect_key( $partial, $allowed_settings );
 
+		// Auto-update isGTGDefault to false when isEnabled changes, but only if isGTGDefault is currently true.
+		if ( isset( $updated['isEnabled'] ) && true === $settings['isGTGDefault'] ) {
```

Comment:
`$updated['isEnabled']` will always be set here because it includes the saved + default value. I think this is meant to use `$partial`. Do we need this in `merge` though? `Setting::on_change` may be more appropriate.